src/syngen/ml/processors/processors.py

```diff
@@ -0,0 +1,368 @@
+            }
+            return output
+        else:
+            return input_data
+
+    @staticmethod
+    def _restore_empty_values(df: pd.DataFrame):
+        """
+        Restore the empty dictionary values in the dataframe for nested fields
+        """
+        df = df.astype(object)
+        for i, row in df.iterrows():
+            for pos_col, col in enumerate(df.columns.to_list()):
+                if pd.isna(df.iloc[i, pos_col]):
+                    columns_set = set(df.columns)
+                    columns_set.remove(col)
+                    nested_fields = [
+                        "".join(i.split(f"{col}.")) for i in columns_set if f"{col}." in i
+                    ]
+
+                    if nested_fields:
+                        df.at[i, col] = dict()
+        return df
+
+    @staticmethod
+    def _load_generated_data(
+            path_to_generated_data: str,
+            *args
+    ) -> pd.DataFrame:
+        """
+        Load generated data from the predefined path
+        """
+        data, schema = DataLoader(path_to_generated_data).load_data()
+        return data
+
+    def _post_process_generated_data(
+            self,
+            data: pd.DataFrame,
+            flattening_mapping: Dict,
+            duplicated_columns: List
+    ) -> pd.DataFrame:
+        """
+        Postprocess the generated data
+        """
+        for old_column, new_columns in flattening_mapping.items():
+            data[new_columns] = self._restore_empty_values(data[new_columns])
+            data[old_column] = data[new_columns]. \
+                apply(lambda row: unflatten_list(row.to_dict(), "."), axis=1)
+            data[old_column] = data[old_column]. \
+                apply(lambda row: self._remove_none_from_struct(row))
+            dropped_columns = set(i for i in new_columns if i not in duplicated_columns)
+            data.drop(dropped_columns, axis=1, inplace=True)
+        return data
+
+    def run(self):
+        """
+        Launch the postprocessing of generated data,
+        and save the processed data to the predefined path
+        """
+        metadata = self._fetch_metadata()
+
+        if os.path.exists(self.path_to_flatten_metadata):
+            logger.info("Start postprocessing of the generated data")
+            for table in metadata.keys():
+                if table == "global":
+                    continue
+                flatten_metadata = self._fetch_flatten_config(table)
+                flattening_mapping = flatten_metadata.get("flattening_mapping")
+                duplicated_columns = flatten_metadata.get("duplicated_columns")
+                order_of_columns = flatten_metadata.get("order_of_columns")
+                path_to_generated_data = (f"{PATH_TO_MODEL_ARTIFACTS}/tmp_store/"
+                                          f"{slugify(table)}/merged_infer_{slugify(table)}.csv")
+                data = self._load_generated_data(path_to_generated_data, table)
```

Comment:
It seems that argument table is redundant.

Reply:
It's not redundant. It sends to the function for the Enterprise version